php-checks/src/main/java/org/sonar/php/checks/WeakSSLProtocolCheck.java

```diff
@@ -0,0 +1,161 @@
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+import static org.sonar.php.checks.utils.CheckUtils.isStringLiteralWithValue;
+
+@Rule(key = WeakSSLProtocolCheck.KEY)
+public class WeakSSLProtocolCheck extends PHPVisitorCheck {
+  public static final String KEY = "S4423";
+
+  private static final String STREAM_CONTEXT_CREATE = "stream_context_create";
+  private static final String STREAM_SOCKET_ENABLE_CRYPTO = "stream_socket_enable_crypto";
+
+  private static final Map<String, List<String>> WEAK_PROTOCOLS = ImmutableMap.of(
+    STREAM_CONTEXT_CREATE, Arrays.asList(
+      "STREAM_CRYPTO_METHOD_ANY_CLIENT",
+      "STREAM_CRYPTO_METHOD_ANY_SERVER",
+      "STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT",
+      "STREAM_CRYPTO_METHOD_TLSv1_0_SERVER",
+      "STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT",
+      "STREAM_CRYPTO_METHOD_TLSv1_1_SERVER"),
+    STREAM_SOCKET_ENABLE_CRYPTO, Arrays.asList(
+      "STREAM_CRYPTO_METHOD_SSLv2_CLIENT",
+      "STREAM_CRYPTO_METHOD_SSLv3_CLIENT",
+      "STREAM_CRYPTO_METHOD_SSLv23_CLIENT",
+      "STREAM_CRYPTO_METHOD_ANY_CLIENT",
+      "STREAM_CRYPTO_METHOD_TLS_CLIENT",
+      "STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT",
+      "STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT",
+      "STREAM_CRYPTO_METHOD_SSLv2_SERVER",
+      "STREAM_CRYPTO_METHOD_SSLv3_SERVER",
+      "STREAM_CRYPTO_METHOD_SSLv23_SERVER",
+      "STREAM_CRYPTO_METHOD_ANY_SERVER",
+      "STREAM_CRYPTO_METHOD_TLS_SERVER",
+      "STREAM_CRYPTO_METHOD_TLSv1_0_SERVER",
+      "STREAM_CRYPTO_METHOD_TLSv1_1_SERVER"
+    )
+  );
+
+  private static final String MESSAGE = "Change this code to use a stronger protocol";
+  private AssignmentExpressionVisitor assignmentExpressionVisitor;
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    assignmentExpressionVisitor = new AssignmentExpressionVisitor(context().symbolTable());
+    tree.accept(assignmentExpressionVisitor);
+    super.visitCompilationUnit(tree);
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    String functionName = CheckUtils.getFunctionName(tree);
+    if (STREAM_CONTEXT_CREATE.equals(functionName)) {
+      SeparatedList<ExpressionTree> arguments = tree.arguments();
+      if (!arguments.isEmpty()) {
+        checkSSLConfig(arguments.get(0));
+      }
+    }
+    if (STREAM_SOCKET_ENABLE_CRYPTO.equals(functionName)) {
+      SeparatedList<ExpressionTree> arguments = tree.arguments();
+      if (arguments.size() > 2) {
+        checkWeakProtocol(getAssignedValue(arguments.get(2)), STREAM_SOCKET_ENABLE_CRYPTO);
+      }
+    }
+    super.visitFunctionCall(tree);
+  }
+
+  private void checkSSLConfig(ExpressionTree expressionTree) {
+    ExpressionTree param = getAssignedValue(expressionTree);
+    if (!param.is(Tree.Kind.ARRAY_INITIALIZER_BRACKET, Tree.Kind.ARRAY_INITIALIZER_FUNCTION)) {
+      return;
+    }
+    ArrayInitializerTree config = (ArrayInitializerTree) param;
+    getProperty(config, "SSL")
```

Comment:
you're using `config` just here, so it would be more readable to pass directly `(ArrayInitializerTree) param` and drop the local variable
and maybe rename `param` to `config` to suggest what we expect from it to be

Reply:
good point, I will rename it and drop the local variable.